src/app/site/checkout/checkout-page/checkout-page.component.ts

```diff
@@ -85,28 +95,86 @@ export class CheckoutPageComponent implements OnInit {
       });
 
     this.checkoutService.isValid$.subscribe(r => this.disabled = (!r && !this.soldOuts));
+    if (!this.authService.userDetails.userId) {
+      this.showEarnPointLabel = false;
+      this.earnedLoyaltyPoint = 0;
+    } else {
+      this.showEarnPointLabel = true;
+      this.checkoutService.loyaltyGroups.subscribe(data => this.loyaltyGroups = data);
+      this.checkoutService.addPointArray.subscribe(data => this.addPointArray = data[0].add_point);
+    }
+
+    // this.checkoutService.setPaymentType(this.paymentType.cash);
+    this.checkoutService.setPaymentType(this.paymentType[this.selectedPaymentType]);
   }
 
   changePaymentType(data) {
     this.usedBalance = 0;
     this.usedLoyaltyPoint = 0;
+    this.selectedPaymentType = data;
 
     switch (data) {
+      case this.paymentType.cash: {
+        this.checkoutService.setPaymentType(data);
+      }
+        break;
       case this.paymentType.balance: {
         this.usedBalance = this.balanceValue;
         this.checkoutService.setPaymentType(data);
       }
         break;
       case this.paymentType.loyaltyPoint: {
-        this.usedLoyaltyPoint = this.loyaltyPoint;
+        this.usedLoyaltyPoint = this.loyaltyPoint * this.loyaltyValue;
         this.checkoutService.setPaymentType(data);
       }
         break;
     }
+
+    this.calculateEarnPoint();
   }
 
-  showDiscountLabel(data) {
+  setCostLabel(data) {
     this.showCostLabel = data;
+    this.calculateEarnPoint();
+  }
+
+  calculateEarnPoint() {
+    let scoreArray;
+    let maxScore;
+    let customer_loyaltyGroup;
+    let valid_loyaltyGroups;
+
+    if (!this.authService.userDetails.userId) {
+      this.earnedLoyaltyPoint = 0;
+      this.showEarnPointLabel = false;
+      return;
+    }
+
+    if (this.selectedPaymentType === 2)
+      this.earnedLoyaltyPoint = 0;
+
+    else if (this.showCostLabel) {
+      // calculate earn point
+      this.earnedLoyaltyPoint = Math.floor(this.total / this.system_offline_offer);
+    }
+    else if (!this.showCostLabel) {
+      // calculate earn point in C&C mode
+      valid_loyaltyGroups = this.loyaltyGroups.filter(el => el.min_score <= this.loyaltyPoint);
+
+      if (!valid_loyaltyGroups.length) {
+        scoreArray = this.loyaltyGroups.map(el => el.min_score);
+        maxScore = Math.min(...scoreArray);
+        customer_loyaltyGroup = this.loyaltyGroups.filter(el => el.min_score === maxScore);
+      }
+      else {
+        scoreArray = valid_loyaltyGroups.map(el => el.min_score);
+        maxScore = Math.max(...scoreArray);
+        customer_loyaltyGroup = valid_loyaltyGroups.filter(el => el.min_score === maxScore);
+      }
+      this.earnedLoyaltyPoint = parseInt(this.addPointArray.filter(el => el.name === customer_loyaltyGroup[0].name)[0].added_point)
```

Comment:
if name is changed during process, it's best to set both of them to lower case and then make comparison